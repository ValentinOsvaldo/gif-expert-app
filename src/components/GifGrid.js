import React from "react";
import useFetchGifs from "../hooks/useFetchGifs";
import GifGridItem from "./GifGridItem";
import PropTypes from 'prop-types';

const GifGrid = ({ category }) => {
  const { data: images, loading } = useFetchGifs(category);

  return (
    <>
      <h3 className="animate__animated animate__fadeIn">{category}</h3>

      {loading && <h2 className="animate__animated animate__fadeIn">Loading...</h2>}

      <article className="container">
        {images.map((image) => (
          <GifGridItem key={image.id} {...image} />
        ))}
      </article>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
}

export default GifGrid;
