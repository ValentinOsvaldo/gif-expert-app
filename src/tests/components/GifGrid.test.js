import React from "react";
import { shallow } from "enzyme";
import GifGrid from "../../components/GifGrid";
import useFetchGifs from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe("<GifGrid />", () => {
  const category = "Halo";

  test("Debe hacer match con el snpashot", () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });
    const wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de mostrar items cuando se cargan imagenes useFetchGifs", () => {
    const gifs = [
      {
        id: 1,
        url: "https://locahost:3000/cualquier/cosa.jpg",
        title: "Cualquier cosa",
      },
    ];

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });

    const wrapper = shallow(<GifGrid category={category} />);

    expect( wrapper ).toMatchSnapshot();
    expect(wrapper.find("p").exists()).toBe(false);
    expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );
  });
});
