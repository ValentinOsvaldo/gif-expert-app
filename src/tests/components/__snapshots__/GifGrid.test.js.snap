// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifGrid /> Debe de mostrar items cuando se cargan imagenes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Halo
  </h3>
  <article
    className="container"
  >
    <GifGridItem
      id={1}
      key="1"
      title="Cualquier cosa"
      url="https://locahost:3000/cualquier/cosa.jpg"
    />
  </article>
</Fragment>
`;

exports[`<GifGrid /> Debe hacer match con el snpashot 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Halo
  </h3>
  <h2
    className="animate__animated animate__fadeIn"
  >
    Loading...
  </h2>
  <article
    className="container"
  />
</Fragment>
`;
